### DISCUSSION TITLE
Building model export as FMU

### DISCUSSION THREAD

**Discussion by adahash** [lbl-srg/modelica-buildings#3258] [Q&A]
Dear all,

I'm trying to export a building model as as FMU for other partners. The building model has construction elements (opaque construction components) and they include two layers (insulation board and concrete). The partners would like to examine the influence of the insulation thickness and thermal conductivity. So, these two are given as parameters.
However, I know that opaque construction component consisits of Solid.Generic components, which has state variables (nSta). So, I've been trying to export the FMU with those two parameters but  they do not appear in the list of parameters for the exported FMU.

Are there any workadounds? tipps? 

Thanks in advance

**Comment by JayHuLBL** [comment]
@adahash The `nSta` is a structural parameter. The value change of structural parameter would affect the total number of equations. It cannot be modified after the model has been translated to FMU. See [here](https://stackoverflow.com/questions/35670621/jmodelica-changing-a-loop-iteration-variable-without-re-compiling). If you are using JModelica (deprived), or Optimica, you can change the parameter during compilation call.

For your question of how to change the thickness and conductivity, you may look into the example `Buildings.ThermalZones.Detailed.Validation.BESTEST.Cases6xx.Case600FF` with the insulation thickness and the conductivity propagated as parameter.
